gcloud-java-dns/src/test/java/com/google/gcloud/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,955 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.testing;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.api.services.dns.model.Change;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.ResourceRecordSet;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class LocalDnsHelperTest {
+
+  private static final String RRSET_TYPE = "A";
+  private static final ResourceRecordSet RRSET1 = new ResourceRecordSet();
+  private static final ResourceRecordSet RRSET2 = new ResourceRecordSet();
+  private static final ResourceRecordSet RRSET_KEEP = new ResourceRecordSet();
+  private static final String PROJECT_ID1 = "2135436541254";
+  private static final String PROJECT_ID2 = "882248761325";
+  private static final String ZONE_NAME1 = "my little zone";
+  private static final String ZONE_NAME2 = "another zone name";
+  private static final ManagedZone ZONE1 = new ManagedZone();
+  private static final ManagedZone ZONE2 = new ManagedZone();
+  private static final String DNS_NAME = "www.example.com.";
+  private static final Change CHANGE1 = new Change();
+  private static final Change CHANGE2 = new Change();
+  private static final Change CHANGE_KEEP = new Change();
+  private Map<String, Object> optionsMap;
+  private LocalDnsHelper localDns;
+  private ManagedZone minimalZone = new ManagedZone(); // to be adjusted as needed
+
+  @BeforeClass
+  public static void before() {
+    RRSET1.setName(DNS_NAME);
+    RRSET1.setType(RRSET_TYPE);
+    RRSET1.setRrdatas(ImmutableList.of("1.1.1.1"));
+    ZONE1.setName(ZONE_NAME1);
+    ZONE1.setDescription("");
+    ZONE1.setDnsName(DNS_NAME);
+    ZONE2.setName(ZONE_NAME2);
+    ZONE2.setDescription("");
+    ZONE2.setDnsName(DNS_NAME);
+    RRSET2.setName(DNS_NAME);
+    RRSET2.setType(RRSET_TYPE);
+    RRSET_KEEP.setName(DNS_NAME);
+    RRSET_KEEP.setType("MX");
+    RRSET_KEEP.setRrdatas(ImmutableList.of("255.255.255.254"));
+    RRSET2.setRrdatas(ImmutableList.of("123.132.153.156"));
+    CHANGE1.setAdditions(ImmutableList.of(RRSET1, RRSET2));
+    CHANGE2.setDeletions(ImmutableList.of(RRSET2));
+    CHANGE_KEEP.setAdditions(ImmutableList.of(RRSET_KEEP));
+  }
+
+  @Before
+  public void setUp() {
+    localDns = LocalDnsHelper.create(0L); // synchronous
```

Comment:
should localDns be static and localDns assignment be in a static `@BeforeClass` `before()` method?